File: GridView/VirtualRadGridView/VirtualRadGridViewCS/VirtualRadGridViewSortingFiltering.Core/VirtualGridViewListSource.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;

namespace VirtualRadGridViewSortingFiltering.Core
{
    public class VirtualGridViewListSource : GridViewListSource
    {
        public VirtualGridDataView DataView { get; private set; }

        public VirtualGridViewListSource(GridViewTemplate template)
            : base(template)
        {
        }

        protected override void InsertItem(int index, GridViewRowInfo item)
        {
            base.InsertItem(index, item);
            this.InitializeBoundRow(item, this.DataView.ItemsSource[index]);
        }

        protected override Telerik.WinControls.Data.RadCollectionView<GridViewRowInfo> CreateDefaultCollectionView()
        {
            this.DataView = new VirtualGridDataView(this);
            return this.DataView;
        }
    }
}

File: GridView/VirtualRadGridView/VirtualRadGridViewSortingFilteringAfterQ3_2015/VirtualRadGridViewSortingFilteringAfterQ3_2015VB/VirtualRadGridViewSortingFiltering.Core/VirtualGridViewListSource.vb

Imports System.Collections.Generic
Imports System.Text
Imports Telerik.WinControls.UI

Namespace VirtualRadGridViewSortingFiltering.Core
    Public Class VirtualGridViewListSource
        Inherits GridViewListSource
        Public Property DataView() As VirtualGridDataView
            Get
                Return m_DataView
            End Get
            Private Set(value As VirtualGridDataView)
                m_DataView = Value
            End Set
        End Property
        Private m_DataView As VirtualGridDataView

        Public Sub New(template As GridViewTemplate)
            MyBase.New(template)
        End Sub


        Protected Overrides Sub InsertItem(index As Integer, item As GridViewRowInfo)
            MyBase.InsertItem(index, item)
            Me.InitializeBoundRow(item, Me.DataView.ItemsSource(index))
        End Sub

        Protected Overrides Function CreateDefaultCollectionView() As Telerik.WinControls.Data.RadCollectionView(Of GridViewRowInfo)
            Me.DataView = New VirtualGridDataView(Me)
            Return Me.DataView
        End Function

    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================
